app/models/game.rb

```diff
@@ -8,4 +8,30 @@ class Game < ApplicationRecord
           OR
           (white_player_id IS NOT NULL AND black_player_id IS NULL)')
         }
+
+  class Board
+    attr_reader :grid
+    
+    def initialize(pieces)
+      @pieces = pieces
+      @grid = Array.new(8) { Array.new(8) }
+      fill_grid
+    end
+    
+    private
+    
+    def pieces
+      @pieces
+    end
+    
+    def fill_grid
+      pieces.each do |piece|
+        next if piece.is_captured
```

Comment:
This could be a scope so that the database is filtering the pieces rather than ruby - you could even reuse the name is_captured:

```ruby
scope :is_captured, -> { where(is_captured: true) }
scope :is_not_captured, -> { where(is_captured: false) }
```